src/cdk/stepper/stepper.ts

```diff
@@ -253,11 +257,14 @@ export class CdkStepper {
   }
 
   private _anyControlsInvalid(index: number): boolean {
-    const stepsArray = this._steps.toArray();
-    stepsArray[this._selectedIndex].interacted = true;
+    this._stepsArray[this._selectedIndex].interacted = true;
     if (this._linear) {
-      return stepsArray.slice(0, index).some(step => step.stepControl.invalid);
+      return this._stepsArray.slice(0, index).some(step => step.stepControl.invalid);
```

Comment:
The `index` here may end up being a negative number (it corresponds to the `selectedIndex` which is a part of the public API) which will result in the `some` skipping the last N items of the array (e.g. `[1, 2, 3, 4].slice(0, -1)` is `[1, 2, 3]`).